src/twitchToDiscord.ts

```diff
@@ -0,0 +1,308 @@
+        else
+        {
+
+            //Record that last given userstate ID (specific to bots)
+            if(userState.id)
+            {
+                //Set the message to contain the special value: botUserStateId
+                lastUserStateMsg.userState.botUserStateId = userState.id;
+            }
+            //TODO: this is a paradox you need to solve - find a way to delete the first message that isn't the buffer; the buffer message is *not* in the message cache.
+            else
+            {
+                console.log('Got the userstate message with no ID -_-');
+
+                //[trailing bot message] This is a special method of recording the last twitch message because we need to have a method of removing this message after the fact
+                const twitchMessage = new twitchMsg(msg, self, userState, channel);
+                lastUserStateMsg = twitchMessage;
+
+                //Adding an node intentionally without a discord message because this is the buffer message.
+                const listNode = messageLinkdListInterface.addNode(new conjoinedMsg(undefined, [twitchMessage]));
+                discordTwitchCacheMap.set(twitchMessage, listNode);
+            }
+
+            //before we override the last message, lets make sure we delete this twitch message if required (...this is jank)
+            //We wait until after the userstate message gets a bot message ID
+            if(lastUserStateMsg?.userState.cueForDelete)
+            {
+                twitchDelete(lastUserStateMsg);
+                manageMsgCache(discordTwitchCacheMap.get(lastUserStateMsg));
+            }
+
+            //If we found the twitch message we wanted to connect, we no longer need it in the cache
+            if(twitchMessageSearchCache[msg])
+            {
+
+                const existingNode = twitchMessageSearchCache[msg],
+                    twitchMessage = new twitchMsg(msg, self, userState, channel);
+
+                lastUserStateMsg = twitchMessage;
+
+                existingNode.data.twitchArray.push(twitchMessage);
+                discordTwitchCacheMap.set(twitchMessage, existingNode);
+
+                //Remove this from the cache since we found it
+                delete twitchMessageSearchCache[msg];
+            }
+        }
+    });
+}
+
+//Login to twitch
+//There are a lot of things that were named differently because this command has the potential of using environemnt variables, so this will be painful to look at and I'm sorry XP
+if(configFile.T2S_DEV_SAVE_TOKEN && fs.existsSync('./devTwitchToken'))
+{
+    //If the file exists, use the thing
+    console.log('==Found the token file! (devTwitchToken!) Using that instead of logging in via oauth (if this token is old, delete the file)==');
+    loginToTwitch({ access_token: fs.readFileSync('./devTwitchToken') as unknown as string }, true);
```

Comment:
I again apologize for this chaos that was brought upon ya XP

it would be good to change this so that we're just passing the raw string so you don't suffer through this type casting hoop